docs/spec_draft.md

```diff
@@ -358,6 +359,69 @@ IEEE-754 specification.
 
 [Back to Ops](#index-of-ops)
 
+## stablehlo.concatenate
+
+### Semantics
+
+Concatenates a variadic number of tensors in `inputs` along `dimension`
+dimension in the same order as the given arguments and produces a `result`
+tensor. More formally,
+`result[i0, ..., id, ..., iR-1] = inputs[k][i0, ..., kd, ..., iR-1]`, where:
+  1. `id = d0 + ... + dk-1 + kd`.
+  2. `d` is equal to `dimension`, and `d0`, ... are `d`'th dimension sizes
+     of `inputs`.
+
+### Operands
+
+| Name        | Type                                              |
+|-------------|---------------------------------------------------|
+| `inputs`    | Variadic number of tensors of any supported types |
+| `dimension` | `si64`                                            |
+
+### Results
+
+| Name     | Type                          |
+|----------|-------------------------------|
+| `result` | tensor of any supported types |
+
+### Constraints
+
+  * (C1) All tensors in `inputs` have the same element type.
+  * (C2) All tensors in `inputs` have the same shape except for the size of the
+  `dimension` dimension.
+  * (C3) `inputs` have N tensors where N >= 1.
+  * (C4) `dimension` is non-negative and less than the rank of the tensors in
+  `inputs`.
```

Comment:
In a similar situation for `iota`, we say "0 $\le$ `iota_dimension` $\lt$ `R`, where `R` is the rank of the `result`". Let's say something similar here - that'd be easier to read.